### PR TITLE
Specify the sampled model in `MapDatasetEventSampler`

### PR DESCRIPTION
**Pull request by fabiopintore**
This PR is intended to solve the issue #5202 and it introduces the possibility to print out the name of the model sampled by the `MapDatasetEventSampler`.

### CODE REVIEW

File: gammapy/datasets/simulate.py [hunk 1]
```diff
@@ -21,6 +22,9 @@
 
 __all__ = ["MapDatasetEventSampler", "ObservationEventSampler"]
 
+logging.basicConfig(level=logging.INFO)
```
Comment (adonath) [inline gammapy/datasets/simulate.py:25]: Please remove the configuration statement here. This usually happens on a global call in the user script.

File: gammapy/datasets/tests/test_simulate.py [hunk 2]
```diff
@@ -590,6 +590,43 @@ def test_mde_run(dataset, models, tmp_path):
     assert u.isclose(obs_back.observatory_earth_location.height, LOCATION.height)
 
 
+@requires_data()
+def test_sampler_output(dataset, model_alternative, caplog):
```
Comment (adonath) [inline gammapy/datasets/tests/test_simulate.py:594]: Probably it's better to add the testing of the logging output to one of the existing test. Sampling a whole new event list just for testing the log output is a bit too much.

**Review by adonath — COMMENTED** [review]
Thanks @fabiopintore! I have left some comment...

File: gammapy/datasets/simulate.py (hunk 1 above)
Comment (fabiopintore) [reply gammapy/datasets/simulate.py]: If I remove it, there is no more output from `logger.info`. I converted all the `logger.info` to simple `print`.

Comment (adonath) [reply gammapy/datasets/simulate.py]: No, please keep the logging statements. We do not use print statements anywhere in the Gammapy code. My points was that the configuration should happen in the user scripts, or in you case, move the logger configuration to the test file.

**Review by bkhelifi — APPROVED** [review]
Thanks @fabiopintore ... Looks good for me.

**Pull request merged**